### DISCUSSION TITLE
Windows High Contrast mode support with forced-colors modifier and system colors

### DISCUSSION THREAD

**Discussion by thibaudcolas** [tailwindlabs/tailwindcss#7839] [Ideas]
I’m interested in using Tailwind on a project with [Windows High Contrast mode](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/) (WHCM) support. Now that Google Chrome has [gained forced-colors emulation support](https://developer.chrome.com/blog/new-in-devtools-98/#forced-colors), I’m expecting lots more people will be interested in doing this – and this could be relatively simple to support directly in Tailwind writing plugins.

There are three things involved:

1. The [forced-colors media query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors), which I’d expect to be a straightforward variant, with a definition very similar to [`reducedMotionVariants`](https://github.com/tailwindlabs/tailwindcss/blob/7b4cc36f5ea1f7e208dcd3af3e8288878735f45f/src/corePlugins.js#L169-L172).
2. [system colors](https://www.w3.org/TR/css-color-4/#css-system-colors) keywords in the default color theme. Only some of those are frequently used, so there might be more thinking needed here – or just leave this for people interested in WHCM support to add to their respective themes.
3. The [forced-color-adjust](https://developer.mozilla.org/en-US/docs/Web/CSS/forced-color-adjust) property, which is used to opt certain elements out of forced colors mode.  Again a simple plugin.

---

For now I’ve gone ahead with making my own tiny plugin just for the one project:

```js
/**
 * forced-colors media query for Windows High-Contrast mode support
 * See https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors.
 */
plugin(({ addVariant }) => {
  addVariant('forced-colors', '@media (forced-colors: active)');
}),
```

and adding system colors to our theme. This works well as-is, but since I’m expecting this will become more and more common of a use case, I thought I should propose it here. If this sounds like a good idea but not-for-now I could also make a reusable plugin with all of the above.

**Comment by thibaudcolas** [comment]
Looks like Tailwind 3.1 will support `prefers-contrast` (#8410), so `forced-colors` should be a natural next step.

**Comment by lukewarlow** [comment]
See #11694 and #11931 these two PRs combined with existing variants should allow for better handling of windows themes.

**Comment by lukewarlow** [comment]
Fwiw I completely agree that system colors should be built in to the Tailwind colour palette too. I raised it here https://github.com/tailwindlabs/tailwindcss/discussions/11930
